packages/react/src/components/Forms/FileUploader/index.tsx

```diff
@@ -0,0 +1,347 @@
+                <div className="c__file-uploader__inner__text">{text}</div>
+              )}
+            </>
+          )}
+          <input
+            type="file"
+            name={name}
+            ref={inputRef}
+            onChange={(e) => {
+              if (e.target.files) {
+                onFilesChange([...e.target.files]);
+              } else {
+                onFilesChange([]);
+              }
+            }}
+            multiple={multiple}
+            {...inputProps}
+          />
+        </div>
+      </label>
+    );
+  }
+);
+
+const FileUploaderMono = forwardRef<FileUploaderRefType, FileUploaderProps>(
+  ({ fakeDefaultFiles, ...props }, ref) => {
+    const { t } = useCunningham();
+    const [file, setFile] = useState<File | undefined>(
+      fakeDefaultFiles && fakeDefaultFiles.length > 0
+        ? fakeDefaultFiles[0]
+        : undefined
+    );
+    // This is made to prevent useEffects inside DropZone that depends on `files` to trigger on each render,
+    // doing this preserves the reference of the array.
+    const files = useMemo(() => (file ? [file] : []), [file]);
+    const [hoverDelete, setHoverDelete] = useState(false);
+    const [icon, setIcon] = useState<ReactNode>();
+    const [animateIcon, setAnimateIcon] = useState(true);
+
+    const deleteFile = (e: React.MouseEvent<HTMLElement>) => {
+      setFile(undefined);
+      // This is to prevent opening the browse file window.
+      e.preventDefault();
+      setHoverDelete(false);
+    };
+
+    const updateIcon = () => {
+      if (hoverDelete) {
+        setAnimateIcon(true);
+        setIcon(
+          props.deleteIcon ?? <span className="material-icons">delete</span>
+        );
+        return;
+      }
+      if (file) {
+        setAnimateIcon(false);
+        setIcon(
+          props.fileSelectedIcon ?? (
+            <span className="material-icons">download</span>
+          )
+        );
+        return;
+      }
+      setAnimateIcon(true);
+      setIcon(props.icon);
+    };
+
+    useEffect(() => {
+      updateIcon();
+    }, [file, hoverDelete]);
+
+    return (
+      <>
+        <DropZone
+          {...props}
+          files={files}
+          onFilesChange={(newFiles) => setFile(newFiles[0])}
+          icon={icon}
+          animateIcon={animateIcon}
+          ref={ref}
+        >
+          {file && (
+            <>
+              <div className="c__file-uploader__inner__filename">
+                {file.name}
+              </div>
+              {/* We cannot use Button component for "Delete file" because it is wrapper inside a label, when clicking */}
+              {/* on the label if a button is wrapper inside it triggers the button click too. But in our situation if a file is */}
+              {/* already choose, clicking on the label should trigger the browse file window and not JUST remove the file. ( We */}
+              {/* must click specifically on the button to remove the file ) */}
```

Comment:
```suggestion
              {/* We cannot use Button component for "Delete file" because it is wrapper inside a label, when clicking */}
              {/* on the label if a button is wrapped inside it triggers the button click too. But in our situation if a file has been */}
              {/* already chosen, clicking on the label should trigger the browse file window and not JUST remove the file. ( We */}
              {/* must click specifically on the button to remove the file ) */}
```